python/ray/resource_spec.py

```diff
@@ -0,0 +1,244 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import namedtuple
+import logging
+import multiprocessing
+import os
+
+import ray
+import ray.ray_constants as ray_constants
+
+logger = logging.getLogger(__name__)
+
+
+class ResourceSpec(
```

Comment:
Moving everything into one file because we need to determine all the limits in one place. Otherwise, it becomes hard to calculate the heap limit since i.e., it depends on knowing how large the object store size is.